echidna/core/spectra.py

```diff
@@ -229,3 +229,29 @@ def add(self, spectrum):
         self._data += spectrum._data
         self._raw_events += spectrum._raw_events
         self._num_decays += spectrum._num_decays
+
+    def rebin(self, *args):
+        '''rebin spectra data into a smaller spectra of the same rank whose
+        dimensions are factors of the original dimensions.
+        Adapted from http://wiki.scipy.org/Cookbook/Rebinning
+
+        Args:
+          *args (int): comma seperated values for new bin numbers.
+
+        Raises:
+          ValueError: Number of dimensions incorrect for spectra
+          ValueError: Old bins/ New bins must be integer
+        '''
+        shape = self._data.shape
+        len_shape = len(shape)
+        if len_shape != len(args):
+            raise ValueError("Number of dimensions incorrect")
+        for i in range(len_shape):
+            if shape[i] % args[i] != 0:
+                raise ValueError("Old bins/ New bins must be integer")
+        factor = numpy.asarray(shape)/numpy.asarray(args)
+        evList = ['self._data.reshape('] + \
+            ['args[ % d],factor[%d],' % (i, i) for i in range(len_shape)] + \
+            [')'] + ['.sum(%d)' % (i+1) for i in range(len_shape)] + \
+            ['/factor[ % d]' % i for i in range(len_shape)]
+        self._data = eval(''.join(evList))
```

Comment:
I think after `self._data` has been rebinned, the relevant members that describe the binning should be updated. I.e. if `self._data` is rebinned along the energy axis, then `self._energy_bins` and `self._energy_width` should be updated accordingly.